src/modules/DistributionManager.sol

```diff
@@ -0,0 +1,535 @@
+import {Ownable} from "@solady/contracts/auth/Ownable.sol";
+
+/// @title DistributionManager
+/// @notice Abstract contract that orchestrates the entire distribution process
+/// @dev Consolidates all distribution logic and provides hooks for module integration
+abstract contract DistributionManager is IDistributionModule, ReentrancyGuard, Pausable, Ownable {
+    using SafeERC20 for IERC20;
+
+    error ZeroAddress();
+    error InvalidCycleLength();
+    error InvalidDivisor();
+    error DistributionNotResolved();
+    error InsufficientYield();
+    error NoRecipients();
+    error CycleNotComplete();
+    error OnlyEmergencyAdmin();
+    error NoYieldSource();
+
+    uint256 private constant PRECISION = 1e18;
+
+    address public emergencyAdmin;
+    address public yieldToken;
+    address public yieldSource;
+
+    uint256 public cycleLength;
+    uint256 public yieldFixedSplitDivisor;
+    uint256 public lastDistributionBlock;
+    uint256 public cycleNumber;
+    uint256 public totalYieldCollected;
+    uint256 public lastCollectionBlock;
+
+    address[] public recipients;
+    uint256[] public currentVotes;
+    uint256 public totalVotes;
+
+    mapping(uint256 => DistributionState) public distributionHistory;
+
+    event YieldCollected(address indexed source, uint256 amount, uint256 blockNumber);
+    event YieldValidated(uint256 totalYield);
+
+    modifier onlyEmergencyAdmin() {
+        if (msg.sender != emergencyAdmin && msg.sender != owner()) {
+            revert OnlyEmergencyAdmin();
+        }
+        _;
+    }
+
+    /// @notice Initializes the distribution manager
+    /// @param _yieldToken Address of the yield token
+    /// @param _yieldSource Address of the yield source
+    /// @param _cycleLength Initial cycle length in blocks
+    /// @param _yieldFixedSplitDivisor Initial fixed split divisor
+    function __DistributionManager_init(
+        address _yieldToken,
+        address _yieldSource,
+        uint256 _cycleLength,
+        uint256 _yieldFixedSplitDivisor
+    ) internal {
+        if (_yieldToken == address(0)) revert ZeroAddress();
+        if (_yieldSource == address(0)) revert ZeroAddress();
+        if (_cycleLength == 0) revert InvalidCycleLength();
+        if (_yieldFixedSplitDivisor == 0) revert InvalidDivisor();
+
+        yieldToken = _yieldToken;
+        yieldSource = _yieldSource;
+        cycleLength = _cycleLength;
+        yieldFixedSplitDivisor = _yieldFixedSplitDivisor;
+        lastDistributionBlock = block.number;
+
+        _initializeOwner(msg.sender);
+    }
+
+    /// @inheritdoc IDistributionModule
+    function distributeYield() external override nonReentrant whenNotPaused {
+        (bool canDistribute,) = validateDistribution();
+        if (!canDistribute) revert DistributionNotResolved();
+
+        // Hook for token minting before distribution
+        _mintTokensBeforeDistribution();
+
+        // Collect yield
+        uint256 totalYield = _collectYield();
```

Comment:
this whole path needs to be simplified, the entire flow is redudant, reference how it is implemented at breadchain and use the base token methods